File: easy/3959_check_good_integer/Solution.java
class Solution {
  public boolean checkGoodInteger(int n) {
    int nCopy = n;

    int digitSum = 0;
    int squareSum = 1;

    while (nCopy != 0) {
      final int digit = nCopy % 10;
      nCopy /= 10;

      digitSum += digit;
      squareSum += digit * digit;
    }

    return squareSum - digitSum >= 50;
  }
}

File: easy/3959_check_good_integer/solution.py
class Solution:
    def checkGoodInteger(self, n: int) -> bool:
        digitSum = 0
        squareSum = 1

        while n:
            digit = n % 10
            n //= 10

            digitSum += digit
            squareSum += digit * digit

        return squareSum - digitSum >= 50
